### PR TITLE
Adding Salesforce Lightning docs

### PR DESCRIPTION
**Pull request by mattgoldspink**
This is an initial version of adding docs for Salesforce's Lightning platform. Original docs are here: https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/intro_framework.htm

Let me know if I've missed anything for this to be contributed.

### CODE REVIEW

File: docsets/Salesforce_Lightning/docset.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "Salesforce Lightning",
+    "version": "40.0",
+    "archive": "Salesforce_Lightning.tgz",
```
Comment (Kapeli) [inline docsets/Salesforce_Lightning/docset.json]: There's no `Salesforce_Lightning.tgz` in the main folder. You need to add it. Also remove `Sample_Docset.tgz`.

**Comment by Kapeli** [comment]
All good otherwise.

File: docsets/Salesforce_Lightning/docset.json (hunk 1 above)
Comment (mattgoldspink) [inline docsets/Salesforce_Lightning/docset.json]: Good catch! Should be fixed now

**Comment by Kapeli** [comment]
Thank you! 👍

**Pull request merged**